scripts/kconfig/kconfigfunctions.py

```diff
@@ -601,12 +601,20 @@ def dt_has_compat(kconf, _, compat):
 def dt_compat_enabled(kconf, _, compat):
     """
     This function takes a 'compat' and returns "y" if we find a status "okay"
-    compatible node in the EDT otherwise we return "n"
+    compatible node, or a status "disabled" + "zephyr,deferred-init" compatible
```

Comment:
I like the "available" term, I think it's more descriptive. However, I'm hesitant to introduce it directly into the DT API because devicetree doesn't have such a concept. It's really a Zephyr thing (it will be strictly tied to the special zephyr,deferred-init property). I don't even like the zephyr,deferred-init property, it exists because Zephyr doesn't have a better compile-time configuration language (Kconfig does not play well with device instances...).

Let's discuss it today because the new terminology will propagate to many places (see 2nd point in open points).